### PR TITLE
[Path] Feature/chipbreak

### LINKED ISSUE #6747: [Feature Request] Implement Chip-breaking drill cycle (G73)
### Is there an existing request for this?

- [X] I have searched the existing requests

### Forums discussion

None

### Subproject(s) affected?

Path

### Idea description

LinuxCNC now support G73 chip breaking.  This is a bit like peck drilling but doesn't fully retract between pecks.  It just retracts a small amount to break off the chip.

http://linuxcnc.org/docs/html/gcode/g-code.html#gcode:g73

### Anything else?

This can be easily implemented in the drill generator and operation but a translation routine will need to be written for postprocessors that don't support G73  (ie grbl).

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by sliptonic**
Support for LinuxCNC G73 chipbreaking.
Fixes #6747 
If chipbreaking is enabled, Path emits a G73 instead of a G83 for peck drilling.  It's up to the postprocessors to handle the G73.  

With chipbreaking, instead of retracting the drill all the way out of the hole to clear chips, the bit is retracted a very small amount to break the chip.

Postprocessors that don't do canned cycles will have to translate the G73 into a a series of G0/G1 moves much as they do the G83 command now.  This translation has not been done yet.

Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [ ]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [ ]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [ ]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [ ]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [ ]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [ ]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [ ]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the issue ID number from our [Issues database](https://github.com/FreeCAD/FreeCAD/issues) in case a particular commit solves or is related to an existing issue. Ex: `Draft: fix typos - fixes #4805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.20 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=56135).

---

### CODE REVIEW

**Comment by berndhahnebach** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7127">pipeline status</a> for feature branch [PR_7127](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7127). Pipeline [578984153](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/578984153) was triggered at [9b97001](https://github.com/FreeCAD/FreeCAD/pull/7127/commits/9b97001254fb16ff82ed8bf8e785b7c2f12f9a74). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/Mod/Path/Generators/drill_generator.py [hunk 1]
```diff
@@ -79,6 +102,12 @@ def generate(edge, dwelltime=0.0, peckdepth=0.0, repeat=1, retractheight=None):
     cmdParams["Z"] = endPoint.z
     cmdParams["R"] = retractheight if retractheight is not None else startPoint.z
 
+    if repeat < 1:
+        raise ValueError("repeat must be 1 or greater")
```
Comment (benj5378) [inline src/Mod/Path/Generators/drill_generator.py:106]: Isn't this code repeat of line 75-76?
```suggestion
```

File: src/Mod/Path/Generators/drill_generator.py [hunk 2]
```diff
@@ -79,6 +102,12 @@ def generate(edge, dwelltime=0.0, peckdepth=0.0, repeat=1, retractheight=None):
...
+
+    if not type(repeat) is int:
+        raise ValueError("repeat value must be an integer")
```
Comment (benj5378) [inline src/Mod/Path/Generators/drill_generator.py:109]: Isn't this code repeat of line 78-79?
```suggestion
```

**Comment by berndhahnebach** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7127">pipeline status</a> for feature branch [PR_7127](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7127). Pipeline [580828762](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/580828762) was triggered at [0a8aa97](https://github.com/FreeCAD/FreeCAD/pull/7127/commits/0a8aa971fcf3992aafe32a5c5e004942d422300d). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by berndhahnebach** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7127">pipeline status</a> for feature branch [PR_7127](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7127). Pipeline [586112368](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/586112368) was triggered at [bc70a53](https://github.com/FreeCAD/FreeCAD/pull/7127/commits/bc70a53bd3bf03a905ad208cfd397eb872d372da). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by berndhahnebach** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7127">pipeline status</a> for feature branch [PR_7127](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7127). Pipeline [589529267](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/589529267) was triggered at [dc4331c](https://github.com/FreeCAD/FreeCAD/pull/7127/commits/dc4331cb91c659938466b0919109dc0bce8bac58). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by berndhahnebach** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7127">pipeline status</a> for feature branch [PR_7127](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7127). Pipeline [589529817](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/589529817) was triggered at [9f7c699](https://github.com/FreeCAD/FreeCAD/pull/7127/commits/9f7c699fb8bb9d51571b9fd7b4ab3b16e75f53a9). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Pull request merged**